File: modules/swagger-core/src/test/scala/JaxBDefaultValueTest.scala
import io.swagger.converter.ModelConverters
import io.swagger.models.properties.IntegerProperty
import io.swagger.models.properties.StringProperty
import models._

import org.junit.runner.RunWith
import org.scalatest.{FlatSpec, Matchers}
import org.scalatest.junit.JUnitRunner

import scala.collection.JavaConverters.mapAsScalaMapConverter

@RunWith(classOf[JUnitRunner])
class JaxBDefaultValueTest extends FlatSpec with Matchers {
  it should "convert a model with Guava optionals" in {
    val schemas = ModelConverters.getInstance().read(classOf[ModelWithJaxBDefaultValues])
    val model = schemas.get("ModelWithJaxBDefaultValues")
    for ((name, property) <- model.getProperties.asScala) {
      name match {
        case "name" =>
          property.asInstanceOf[StringProperty].getDefault should be ("Tony")
        case "age" =>
          property.asInstanceOf[IntegerProperty].getDefault should be (100)
        case _ =>
          fail(s"""Property "${name}" was not expected""")
      }
    }
  }
}

File: modules/swagger-core/src/test/java/io/swagger/JaxBDefaultValueTest.java
package io.swagger;

import io.swagger.converter.ModelConverters;
import io.swagger.oas.models.ModelWithJaxBDefaultValues;
import io.swagger.oas.models.media.IntegerSchema;
import io.swagger.oas.models.media.Schema;
import io.swagger.oas.models.media.StringSchema;
import org.testng.annotations.Test;

import java.util.Map;

import static org.testng.Assert.assertEquals;

public class JaxBDefaultValueTest {

    @Test(enabled = false, description = "convert a model with Guava optionals")
    public void convertModelWithGuavaOptionals() {
        final Map<String, Schema> schemas = ModelConverters.getInstance().read(ModelWithJaxBDefaultValues.class);
        final Map<String, Schema> properties = schemas.get("ModelWithJaxBDefaultValues").getProperties();
        assertEquals(properties.size(), 2);
        assertEquals(((StringSchema) properties.get("name")).getDefault(), "Tony");
        assertEquals((int) ((IntegerSchema) properties.get("age")).getDefault(), 100);
    }
}
